kyo-bench/src/main/scala/kyo/bench/FailureBench.scala

```diff
@@ -0,0 +1,89 @@
+package kyo.bench
+
+class Ex1  extends Exception
+object Ex1 extends Ex1
+
+class Ex2  extends Exception
+object Ex2 extends Ex2
+
+class FailureBench extends Bench.SyncAndFork[Either[Ex1 | Ex2, Int]]:
+    val depth = 10000
+
+    def catsBench() =
+        import cats.effect.*
+
+        def loop(i: Int): IO[Either[Ex1, Either[Ex2, Int]]] =
+            if i > depth then IO.pure(Right(Right(i)))
+            else
+                (i % 4) match
+                    case 0 => IO.pure(Left(Ex1))
+                    case 1 => IO.pure(Right(Left(Ex2)))
+                    case 2 =>
+                        loop(i - 1).map {
```

Comment:
I must be missing something. If `i` starts at `0` and is only decremented, will this ever reach the `i > depth` condition? Is this meant to run forever?

Reply:
My bad, I hadn't noticed my last push had failed. I've just pushed the correct version, this one was actually running a single iteration since it'd start from zero and get a failure on the first loop.